include/foxy/client_session.hpp

```diff
@@ -53,35 +53,26 @@ struct client_session : public session
 
   explicit client_session(boost::asio::io_context& io, session_opts opts = {});
 
-  // async_connect wraps asio::async_connect method and will invoke
-  // asio::async_connect with the specified host and service parameters
-  // async_connect is SSL-aware and will automatically perform the async
-  // handshake for the user, using the SSL context found in the session's opts
+  // async_connect wraps asio::async_connect method and will invoke asio::async_connect with the
+  // specified host and service parameters async_connect is SSL-aware and will automatically perform
+  // the async handshake for the user, using the SSL context found in the session's opts
   //
   template <class ConnectHandler>
-  auto async_connect(
-    std::string host,
-    std::string service,
-    ConnectHandler&&
-      handler) & -> BOOST_ASIO_INITFN_RESULT_TYPE(ConnectHandler,
-                                                  void(
-                                                    boost::system::error_code,
-                                                    boost::asio::ip::tcp::
-                                                      endpoint));
-
-  // async_request will serialize the provided Request to the connected remote
-  // and then use the provided ResponseParser to parse the response
-  // The user is expected to retrieve the underlying message via the Beast
-  // http::parser interface
+  auto
+  async_connect(std::string host, std::string service, ConnectHandler&& handler) & -> return_t<
```

Comment:
prefer `string_view` to `string`, let the implementation decide if and how copies are made.

Reply:
I remembered why I used strings here. It's because I'm a C++14 library so I don't have string view available which only leaves `std::string const&` as the `asio::string_view`.

This is a deeper refactor where I'd have to enable optional `std::string_view` support which is currently out-of-scope but will be an overall larger and deeper refactor.

I'm taking notes on this for later.

Reply:
Use `beast::string_view`

Reply:
Ooh, good point! That's a much better idea lol